src/Api/Controllers/CollectionsController.cs

```diff
@@ -520,6 +520,13 @@ private async Task<CollectionAccessDetailsResponseModel> GetDetails_vNext(Guid i
 
     private async Task<ListResponseModel<CollectionAccessDetailsResponseModel>> GetManyWithDetails_vNext(Guid orgId)
     {
+        var readManyWithDetailsAuthorized =
+            (await _authorizationService.AuthorizeAsync(User, CollectionOperations.ReadManyWithDetails(orgId))).Succeeded;
+        if (!readManyWithDetailsAuthorized)
+        {
+            throw new NotFoundException();
+        }
```

Comment:
Isn't this expressing the same thing as ReadAllWithAccess, which we already check below? ("Details" just means Access.)

And anyway, I was thinking of this - we need to check whether the user has access to these specific collections, which we already have logic for:

```cs
        if (readAllAuthorized)
        {
             // snip
            // early return
        }

        var readAssignedAuthorized = await _authorizationService.AuthorizeAsync(User, assignedOrgCollections, BulkCollectionOperations.ReadWithAccess);
        // throw if not authorized

        return new ListResponseModel<CollectionAccessDetailsResponseModel>(assignedOrgCollections.Select(c =>
            new CollectionAccessDetailsResponseModel(c.Item1, c.Item2.Groups, c.Item2.Users)
            {
                Assigned = true // Mapping from assignedOrgCollections implies they're all assigned
            })
          );
```

Reply:
Yes! I was overcomplicating it, thank you!